dali/operators/image/remap/displacement_filter.h

```diff
@@ -49,7 +50,7 @@ class DisplacementIdentity {
 template <typename Backend,
           class Displacement = DisplacementIdentity,
           bool per_channel_transform = false>
-class DisplacementFilter : public Operator<Backend> {};
+class DisplacementFilter : public StatelessOperator<Backend> {};
```

Comment:
This code smells.... Some of the derived classes are apparently stateful (jitter?)... I don't like the idea of stating one fact in a base and changing it in a derived class. Perhaps the StatelessOperator should be defined not in terms of backend, but in terms of base class, e.g.:
```C++
template <typename Base>
class StatelessOp<Base> : public Base {
/* add trivial imlementation of checkpointing here */
};

template <typename Backend>
using StatelessOperator = StatelessOp<Operator<Backend>>;
```

With that you could do:
```C++
template <typename Backend>
class Water : public StatelessOp<DisplacementFilter<Backend>>;
```

Reply:
@mzient getting rid of StatelessOperator is the first thing on my refactoring todo list. As soon as the checkpointing will work as a whole (hopefully, this week), I'll try to get rid of `StatelessOperator` and either move this to schema or make it the default as you suggest.